### PR TITLE
Add f-strings with `flynt`

### PR DESCRIPTION
**Pull request by DanielNoord**
## Steps

- [x] Write a good description on what the PR does.

## Description

Sorry for the size of this PR but it should be fairly easy to review. It's all changes of strings > f-strings.
This is all done with [flynt](https://github.com/ikamensh/flynt) which I also used to do the same for `pylint`. I checked every change and did not see any problems.
There are some possible f-strings in the codebase which didn't get picked up by `flynt` which I will tackle after this has been merged.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

### CODE REVIEW

**Comment by Pierre-Sassoulas** [comment]
Well we might want to advise user to use flynt instead of pyupgrade then, because pyupgrade is cautious and do not upgrade everything. (https://github.com/PyCQA/pylint/blob/main/doc/whatsnew/2.11.rst#summary----release-highlights)

**Comment by DanielNoord** [comment]
I'm actually not sure what the best option is. This was the first I found on StackOverflow and which also focuses on this specific task. We can suggest both?

File: tests/unittest_regrtest.py [hunk 1]
```diff
@@ -219,16 +219,15 @@ def test_unicode_in_docstring(self) -> None:
...
         from __future__ import unicode_literals
 
         class MyClass(object):
             def method(self):
-                "With unicode : %s "
+                "With unicode : {'’'} "
 
         instance = MyClass()
         """
-            % "\u2019"
```
Comment (Pierre-Sassoulas) [inline tests/unittest_regrtest.py:231]: 👌

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
Great, I did not know about flynt, I'm glad you made me discover that one 👍 We can remove the disable in pylintrc and this is good to go :)

**Comment by DanielNoord** [comment]
@Pierre-Sassoulas Can you test this with `pylint` before merging? My benchmarks are indicating that `pylint` becomes slower with this branch of `astroid`. Could be my computer doing too much at once, but I ran it twice and it was slower on both occasions.

**Comment by Pierre-Sassoulas** [comment]
On my side it's ~=0,3% faster with f-strings, It's close to what I believed would happen.

**Comment by DanielNoord** [comment]
Probably my computer needing weekend then.. 😄

**Comment by DanielNoord** [comment]
Btw, do you normally `squash and merge` or `rebase and merge` when committing PRs?

**Comment by Pierre-Sassoulas** [comment]
Depends on the commits in the merge requests, if there's a pre-commit auto-fix commit or a merge update from master I'll squash, if the commit are clean and well organized (or too big to be squashed) I'll rebase .  Generally someone who does well organized commit will rebase themself so I don't often have to ask for a rebase.

**Pull request merged**